Tests/ResponseSerializationTests.swift

```diff
@@ -266,8 +348,8 @@ class ResponseSerializationTestCase: BaseTestCase {
         XCTAssertNotNil(result.error, "result error should not be nil")
 
         if let error = result.error {
-            XCTAssertEqual(error.domain, Error.Domain, "error domain should match expected value")
-            XCTAssertEqual(error.code, Error.Code.JSONSerializationFailed.rawValue, "error code should match expected value")
+            XCTAssertEqual(error.domain, NSCocoaErrorDomain, "error domain should match expected value")
+            XCTAssertEqual(error.code, 3840, "error code should match expected value")
```

Comment:
Yeah it definitely is an `NSPropertyListReadCorruptError`. Looks like they used the same errors for JSON as they did plists rather than redefining special error codes for JSON. With that said, should we keep the `data.length > 0` check to instead through an actual Alamofire JSON error instead? I think it would make sense in both JSON and plist cases to keep that check. Thoughts?